### PR TITLE
Auth token openapi

### PR DESCRIPTION
**Pull request by jaylees14**
Based on #367, so quite a large diff.

Adds the auth required to each endpoint, if necessary, as well as the overall security scheme that describes the token

### CODE REVIEW

File: src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala [hunk 1]
```diff
@@ -260,9 +266,12 @@ object OpenAPIGenerator {
...
   private def build(root: OpenAPIRoot): OpenAPIFile = {
     val builder = new OpenAPIGenerator(root.name, root.version, root.description)
-    root.services.foreach(builder.addPaths)
+    val securityScheme = root.auth.map {
+      case Auth.Email => "bearerAuth" -> SecurityScheme("http", "bearer", "JWT")
+    }
+    root.services.foreach(builder.addPaths(securityScheme.map { case (name, _) => name }))
     root.auth.foreach(builder.addAuthPaths)
-    builder.toOpenAPI
+    builder.toOpenAPI(securityScheme)
```
Comment (c-harding) [inline src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala]: Can securityScheme be added to the constructor of `OpenAPIGenerator` instead?

File: src/main/scala/temple/generate/target/openapi/ast/Handler.scala [hunk 2]
```diff
@@ -18,6 +19,10 @@ case class Handler(
   override def jsonOptionEntryIterator: IterableOnce[(String, Option[Json])] = Seq(
     "summary"     ~~> Some(summary),
     "description" ~~> when(description.nonEmpty) { description },
+    "security" ~~> security.map { name =>
+      // What a mess...
+      Seq(Map(name -> Seq[String]()))
```
Comment (c-harding) [inline src/main/scala/temple/generate/target/openapi/ast/Handler.scala:24]: ❤️

**Review by c-harding — APPROVED** [review]
lgtm

**Review by c-harding — APPROVED** [review]
very nice now 😄

**Review by Lewiky — APPROVED** [review]
LGTM

**Pull request merged**